packages/engine/src/plugins/Plugin.ts

```diff
@@ -17,9 +18,8 @@ export interface IPlugin {
   description: string
 
   /**
-   * Gets the selected parameter view.
-   * @param param - The parameter with info.
-   * @returns The JSX element or undefined if this plugin does not have a visual element.
+   * Function to generate input options.
    */
-  getSelectedParamView?: (param: ParamWithInfo) => JSX.Element | undefined
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  generateInitialOptions: () => any
```

Comment:
When an input is created, this function is called to give it whatever initial options it should have (e.g. default channel and note)

Reply:
hmm, what happens if a plugin wants to manage multiple input types? Maybe the midi one adds a velocity to `float` or a  pitch -> `Shot<number>`